### DISCUSSION TITLE
pcap fails to write to /nsm/

### DISCUSSION THREAD

**Discussion by jwmasekre** [Security-Onion-Solutions/securityonion#11203] [2.4] [answered]
### Version

2.4.10

### Installation Method

Security Onion ISO image

### Description

other (please provide detail below)

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### Network Traffic Collection

span port

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

Yes, there are additional clues in /opt/so/log/ (please provide detail below)

### Detail

I've been struggling with an issue for the past week, and have just figured out what the source problem is. Simply put, if `/nsm/` doesn't have enough space to write a PCAP to, stenoquery just doesn't write to `/nsm/pcapout` and no logs are generated. I had dropped the `diskfreepercentage` down to test something, and then was unable to pull PCAPs; the error given was simply that the PCAP didn't exist.

I confirmed that the PCAP just wasn't being written by jumping into `so-sensoroni` and manually executing the script, both pointing to `/nsm/pcapout/` and `~`, and while the home directory succeeded, the pcapout just didn't exist.

I really don't know what the right answer to this is (checking to see if the pcap wrote, and if not return an error code?), but at the very least maybe if someone runs into this in the future they can find this discussion and hopefully solve their issue

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by dougburks** [comment, marked as answer]
I've added a Troubleshooting section to the PCAP page in our documentation:
https://docs.securityonion.net/en/2.4/pcap.html#troubleshooting
